### PR TITLE
Rework setting default shell by using chsh.  Do not rely on SQL.

### PR DESCRIPTION
**Pull request by beckhamrryyaann**
### Changes
Based on the request from @chrjorgensen I have reworked the setting of the default shell to use chsh.  SQL is no longer used.  Both chsh and bash have to be installed before code will check for default shell.

Description of change here.

### Checklist

* [X] have tested my change
* [ ] updated relevant documentation
* [X] Remove any/all `console.log`s I added
* [X] eslint is not complaining
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/halcyon-tech/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [X] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

**Comment by worksofliam** [comment]
@beckhamrryyaann Did this conversation happen in a GitHub issue? What was the issue number for that? I am a bit out of the loop!

**Comment by beckhamrryyaann** [comment]
@worksofliam see link to the conversation.  It was on my previous pull request.   
https://github.com/halcyon-tech/vscode-ibmi/pull/1134#discussion_r1136585794

**Review by chrjorgensen — CHANGES_REQUESTED** [review]
Nice work, works like a charm! There is only a minor change needed...

File: src/api/IBMi.ts [hunk 1]
```diff
@@ -624,6 +583,49 @@ export default class IBMi {
...
+              command: `echo $SHELL`
+            });
+
+            if (!commandShellResult.stderr) {
+              let userDefaultShell = commandShellResult.stdout.trim();
+              if (userDefaultShell !== bashShellPath) {
+
+                vscode.window.showInformationMessage(`IBM recommends using bash as your default shell.`, `Set shell to bash?`, `Read More`,).then(async choice => {
+                  switch (choice) { 
+                    case `Set shell to bash?`:
```
[the remark this answers is not in the document]
Comment (beckhamrryyaann) [reply src/api/IBMi.ts]: fixed.

**Comment by chrjorgensen** [comment]
@beckhamrryyaann LGTM!

@worksofliam Any comments on this?

**Comment by worksofliam** [comment]
@chrjorgensen Review incoming this week.

**Comment by worksofliam** [comment]
@beckhamrryyaann I reviewed and I am super happy with this! I really appreciate your PRs!

I just need you to fix the conflict (we had a big PR merged!), then I will retest again, and we should be good to merge it.

Thanks!

**Comment by beckhamrryyaann** [comment]
@worksofliam I have fixed the conflicts.....ready for review.

File: src/api/IBMi.ts [hunk 2]
```diff
@@ -626,6 +585,92 @@ export default class IBMi {
...
+                });
+              }
+            }
+          }
+        } catch (e) {
+          // Oh well...trying to set default shell is not worth stopping for.
+          console.log(e);
+        }
+
+        // Check users default shell.
```
Comment (chrjorgensen) [inline src/api/IBMi.ts:631]: This code is a duplicate of the code block right above - please cleanup.

**Review by chrjorgensen — CHANGES_REQUESTED** [review]
Duplicated code...

File: src/api/IBMi.ts (hunk 2 above)
Comment (beckhamrryyaann) [reply src/api/IBMi.ts]: Let’s  try this again.  Duplicate code removed. sorry about that.  Ready for review.

**Review by chrjorgensen — APPROVED** [review]
Looks good to me and works as expected.

**Pull request merged**